src/brushes/eraser_brush.class.js

```diff
@@ -0,0 +1,361 @@
+          clipPath: clipObject,
+          dirty: true,
+          eraser: true
+        });
+      },
+
+      /**
+       * On mouseup after drawing the path on contextTop canvas
+       * we use the points captured to create an new fabric path object
+       * and add it to every intersected erasable object.
+       */
+      _finalizeAndAddPath: function () {
+        var ctx = this.canvas.contextTop, canvas = this.canvas;
+        ctx.closePath();
+        if (this.decimate) {
+          this._points = this.decimatePoints(this._points, this.decimate);
+        }
+        var pathData = this._points && this._points.length > 1 ? this.convertPointsToSVGPath(this._points).join("") : "M 0 0 Q 0 0 0 0 L 0 0";
+        if (pathData === "M 0 0 Q 0 0 0 0 L 0 0") {
+          // do not create 0 width/height paths, as they are
+          // rendered inconsistently across browsers
+          // Firefox 4, for example, renders a dot,
+          // whereas Chrome 10 renders nothing
+          this.canvas.requestRenderAll();
+          return;
+        }
+
+        var path = this.createPath(pathData);
+        canvas.clearContext(canvas.contextTop);
+        canvas.fire("before:path:created", { path: path });
+
+        this.restoreCanvasImage(canvas, path, 'backgroundImage', 'setBackgroundImage');
+        this.restoreCanvasImage(canvas, path, 'overlayImage', 'setOverlayImage');
+        this.restoreCanvasColor(canvas, path, 'backgroundColor', 'setBackgroundColor');
+        this.restoreCanvasColor(canvas, path, 'overlayColor', 'setOverlayColor');
+        var _this = this;
+        canvas.forEachObject(function (obj) {
+          if (obj.erasable && obj.intersectsWithObject(path)) {
+            _this._addPathToObjectEraser(obj, path);
+          }
+        });
+        canvas.requestRenderAll();
+        path.setCoords();
+        this._resetShadow();
+
+        // fire event 'path' created
+        canvas.fire("path:created", { path: path });
+      }
+    }
+  );
+
+  /**
+   * Used by @class fabric.EraserBrush
+   * Can be used regardless of @class fabric.EraserBrush to create an inverted clip path made of strokes (=unclosed paths)
+   * @private
+   * @class fabric.EraserPath
+   * @extends fabric.Rect
+   */
+  fabric.EraserPath = fabric.util.createClass(fabric.Rect, {
+
+    type: 'eraserPath',
+
+    stateProperties: fabric.Object.prototype.stateProperties.concat('_paths'),
+
+    cacheProperties: fabric.Object.prototype.cacheProperties.concat('_paths'),
+
+    _paths: [],
+
+    initialize: function (paths, options) {
+      this.callSuper('initialize', Object.assign(options || {}, {
+        originX: 'center',
+        originY: 'center'
+      }));
+      this._paths = paths || [];
+      this._paths.forEach(function (p) {
+        p.path.set({ globalCompositeOperation: "destination-out" });
+      })
+    },
+
+    setParent: function (parent) {
+      this.set({
+        width: parent.width,
+        height: parent.height,
```

Comment:
When `parent` is `fabric.Path` it behaves a bit differently. `width` and `height` aren't exactly correct so the clip path removes the edges. How can I handle this correctly?